src/requirements.cpp

```diff
@@ -125,7 +125,11 @@ std::string item_comp::to_string( const int batch, const int avail ) const
     const int c = std::abs( count ) * batch;
     const auto type_ptr = item::find_type( type );
     if( type_ptr->count_by_charges() ) {
-        if( avail > 0 ) {
+        if( avail == INT32_MAX ) {
+            return string_format( npgettext( "requirement", "%2$d %1$s of infinite", "%2$d %1$s of infinite",
+                                             c ),
+                                  type_ptr->nname( c ), c );
```

Comment:
```suggestion
                                  type_ptr->nname( 1 ), c );
```
Singular form is always used for items counted by charges